File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> queue = new PriorityQueue<>((p0, p1) -> (p1[0]*p1[0] + p1[1]*p1[1]) -(p0[0]*p0[0] + p0[1]*p0[1]));

        for (int[] point : points) {
          queue.offer(point);
          if (queue.size() > k) {
            queue.poll();
          }
        }

        int[][] res = new int[k][];
        
        int count = 0;
        while (!queue.isEmpty()) {
          res[count++] = queue.poll();
        }

        return res;
    }
}

File: 1014-k-closest-points-to-origin/1014-k-closest-points-to-origin.cpp
class Solution {
public:

    struct Compare{
      bool operator() (const vector<int>& v1, const vector<int>& v2) {
        return (v1[0] * v1[0] + v1[1] * v1[1]) < (v2[0] * v2[0] + v2[1] * v2[1]); 
      }
    };

    vector<vector<int>> kClosest(vector<vector<int>>& points, int k) {
        priority_queue<vector<int>, vector<vector<int>>, Compare> pq;

        for (vector<int> point : points) {
          pq.push(point);

          if (pq.size() > k) {
            pq.pop();
          }
        }

        vector<vector<int>> res;

        while (!pq.empty()) {
          res.push_back(pq.top());
          pq.pop();
        }

        return res;
    }
};